### DISCUSSION TITLE
alternate match colors when matches are adjacent

### DISCUSSION THREAD

**Discussion by AeliusSaionji** [BurntSushi/ripgrep#2679] [General]
I was in the process of iteratively building out a regex to match an IPv4 address, and even before I added repetition, I noticed rg already seemed to match against three of the four segments of IPv4.

`rg '\d+\.'`

![image](https://github.com/BurntSushi/ripgrep/assets/4342746/8742e45a-382e-4161-bb86-450f838185d9)

This momentarily confused me, until I realized these were three separate side by side matches.

Perhaps ripgrip could introduce alternate colors when two matches are side by side?
I think that would be a nice feature.

thanks!

**Comment by BurntSushi** [comment]
I'm going to pass on this one. It sounds like a nice idea, and I don't necessarily disagree that it is, but if you sit down and think about how it might be implemented, it's pretty complex. Users will absolutely demand some kind of control over which colors are used, and that will complicate the UX. There's also the possibility that end users won't understand what the different colors mean, and might think there is some significance to them. That might in turn imply that this needs to be a non-default opt-in feature, which reduces its effectiveness somewhat since you need to know to opt into it.

One thing worth mentioning here is that if you are confused about how ripgrep comes up with the match it does, you can pass the `-o/--only-matching` flag to put each individual match on its own line. This should clear things up quite a bit and is I think better than trying to fiddle with the coloring.
